addons/survey/static/src/js/survey_form.js

```diff
@@ -1109,6 +1112,15 @@ publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend(SurveyPreloa
                 $(dependingQuestion).find('textarea').val('');
             }
         }
+        // Sections visibility
+        if (this.options.questionsLayout !== 'one_page'){
+            return;
+        }
+        const sections = document.querySelectorAll('.js_section_wrapper');
```

Comment:
Instead of always iterating over all sections, what about only iterating on the questions whose visibility has to be updated and hide the parent section if all questions of the said section are hidden?

Concretely,
```
For each question to hide/show:
    Fetch the question in the DOM
    Fetch the parent section (you can use the `closest` function)
    If there is a section:
         Select all questions of that section
         Toggle the `d-none` class based on the visibility of the questions you selected
```

We will then avoid iterating over sections unnecessarily. It should be faster. We could even do everything within the same loop (see above).

Note: The function `_applyConditionalQuestionsVisibility` is called with a variable named `conditionalQuestionsToRecomputeVisibility`. So, I assume that `questionsIds` is a list containing the ids of the questions whose visibility has to be updated.